.github/workflows/journey-map.yml

```diff
@@ -0,0 +1,47 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches:
+      - main
+    paths:
+      - 'journey-map/**'
+      - 'lambdas/process-journey-event/src/main/resources/statemachine/journey-maps/**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
```

Comment:
I'm not super good with GHA. Did you have to create this environment, or does it already exist due to it being github-pages? 
The value for the URL comes from the deployment step, and seems to not be configurable. I guess I'm wondering why we need this environment at all? What would happen if we didn't have it at all. Would the URL output by deploy work? Would the deploy fail for some reason?

Reply:
I think this is referencing a github 'environment', as opposed to environment variables as such.

It links to the settings, which allows us to place protection on who can deploy to it etc., as well as secrets/variables (https://github.com/govuk-one-login/ipv-core-back/settings/environments/1454443961/edit) - I'm not sure what exactly would happen if we omitted it!

Reply:
I think `github-pages` is created automatically when you configure github pages - at the very least I didn't create it manually.

Reply:
Ah ok, that makes sense. There's the `actions/configure-pages` step which I suppose did the initial configuration. I can see with the API that the env was created at `"created_at": "2023-10-12T15:42:40Z"`, which I'm guessing was the first run of the workflow maybe.